pkg/gitversion/gitversion.go

```diff
@@ -442,3 +466,52 @@ func workTreeIsDirty(repo *git.Repository) (bool, error) {
 	}
 	return len(output) > 0, nil
 }
+
+func walkRemoteParents(repo *git.Repository, commit *object.Commit, f func(*object.Commit) error) error {
+	queue := []*object.Commit{commit}
+	fetchDepth := 4
+	seen := map[plumbing.Hash]struct{}{}
+	for i := 0; len(queue) > i; i++ {
+		commit := queue[i]
+		if _, ok := seen[commit.Hash]; ok {
+			continue
+		} else {
+			seen[commit.Hash] = struct{}{}
+		}
+		err := f(commit)
+		if errors.Is(err, storer.ErrStop) {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		for _, p := range commit.ParentHashes {
+			pCommit, err := repo.CommitObject(p)
+			if errors.Is(err, plumbing.ErrObjectNotFound) {
+				// The parent is missing, so lets fetch it.
+				fetchDepth *= 2
+				err = repo.Fetch(&git.FetchOptions{
+					RemoteName: "origin",
+					Depth:      fetchDepth,
+					RefSpecs: []config.RefSpec{config.RefSpec(
+						fmt.Sprintf("%[1]s:%[1]s", p.String()),
+					)},
+					Tags:     git.TagFollowing,
```

Comment:
Does this fetch commits only or it fetches tags from the remote also if these commits are tagged on the remote?

Reply:
`Tags: git.TagFollowing` will fetch commits and any tag applied to them.